src/main/java/org/candlepin/subscriptions/resteasy/EnumParamConverterProvider.java

```diff
@@ -53,34 +75,85 @@
      */
     public static class EnumParamConverter<T> implements ParamConverter<T> {
 
-        private final Map<String, T> stringValueMap = new HashMap<>();
         private final Class<T> clazz;
+        private Map<String, T> keyValuePairs = new HashMap<>();
 
-        public EnumParamConverter(Class<T> clazz) {
-            this.clazz = clazz;
-            for (T value : clazz.getEnumConstants()) {
-                String stringValue = value.toString();
-                stringValueMap.put(stringValue, value);
+        public EnumParamConverter(Class<T> className) {
+            this.clazz = className;
+
+            if (Objects.nonNull(className.getEnumConstants())) {
+                keyValuePairs = Arrays
+                    .stream(className.getEnumConstants())
+                    .collect(Collectors.toMap(T::toString, value -> value));
             }
         }
 
-        @Override
+        public Class<T> getClazz() {
+            return clazz;
+        }
+
+        public Map<String, T> getKeyValuePairs() {
+            return keyValuePairs;
+        }
+
         public T fromString(String value) {
-            if (value == null) {
+            if (Objects.isNull(value)) {
                 return null;
             }
-            if (stringValueMap.containsKey(value)) {
-                return stringValueMap.get(value);
+
+            Optional<Map.Entry<String, T>> result = keyValuePairs
+                .entrySet()
+                .stream()
+                .filter(kv -> value.equals(kv.getKey()))
+                .findFirst();
+
+            if (result.isPresent()) {
+                return result.get().getValue();
             }
```

Comment:
Would be more performant (untested suggestion):
```suggestion
            T result = keyValuePairs.get(value);
            if (result != null) {
                return result;
            }
```